src/rest/Ratings/addRating.ts

```diff
@@ -0,0 +1,49 @@
+import type { Request, Response } from "express";
+import { addRating } from "../../db/Rating/rating";
+import { ObjectId } from "mongodb";
+import type { Rating } from "../../db/types";
+
+export const addRatingToPlace = async (
+	req: Request<unknown, unknown, Partial<Rating>>,
+	res: Response
+): Promise<void> => {
+	const rating = req.body;
+
+	// userId and placeId are mandatory fields
+	if (!rating.userID) {
+		res.status(400).json({ error: "You must provide a userId" });
+		return;
+	}
+
+	if (!rating.placeID) {
+		res.status(400).json({ error: "You must provide a placeId" });
+		return;
+	}
+
+	// need to check that comment is a string otherwise the linter yells at me
+	if (typeof rating.comment !== "string") {
+		res.status(400).json({ error: "comment must be a string" });
+		return;
+	}
+
+	if (rating.comment) {
+		rating.comment = { _id: new ObjectId(), comment: rating.comment };
+	}
```

Comment:
The way I designed this is kind of strange and I realize now that I should have explained this better, sorry about that! The intention of my design is that when comment is first supplied, it's just a string and the objectID is manually created since the idea is that `Comment` is a 'sub-collection' of `Rating`. Normally, `_id` is automatically created automatically when a new document is inserted into the collection, but I'm doing that manually here to mimic a collection inside another collection (Comment inside of Rating). There is probably a better way to do this, but I was going to improve this at a later date because there must be a better way to do this. So, this chunk of code converts `rating.comment` to be a `Comment` object instead of a string (which is the original state). Not sure if that made sense, lmk if I need to explain that better. 

Original request body example:
```
{
    "userID": "617affe7f76be1c791c4c23f",
    "placeID": "fakeid2",
    "fontReadability": 3,
    "staffHelpfulness": 1,
    "navigability": 1,
    "guideDogFriendly": 1,
    "comment": "this is okay"
}
```

Rating that will be inserted after the code is executed:
```
{
    "userID": "617affe7f76be1c791c4c23f",
    "placeID": "fakeid2",
    "fontReadability": 3,
    "staffHelpfulness": 1,
    "navigability": 1,
    "guideDogFriendly": 1,
    "comment": {
          _id: "618affe7f86be1c791c4c88f",
          comment: "this is okay"
     }
}
```

Reply:
I'm not a huge fan of this solution. If this is what we're going to stick with, maybe we should have an alternate type, say, `RatingReqBody`, that has the `comment` field defined to be a string that would be used alongside the `Rating` type proper. Then we would use `RatingReqBody` up until the very point where the rating needs to be added to the database, then build a `Rating` from the rating request body and add that to the database. 

If this were a more strongly typed language, something like this would normally not be allowed. I don't think we should keep such a workaround around for long. That's why I suggested a new field called `commentText` to be used to avoid a scenario like the one above where we use two very similar types, or pretend that the type of a field is actually another.

Reply:
You know what, now that I'm thinking about it, I don't think comment needs to be a Comment object, it just needs to be a string. I refactored the db schema so many times while writing the API that I think I forgot to rethink how comments would work with the latest changes I made, and now that I'm looking at it holistically, I think I can get rid of that completely. Since Rating has its own id, the placeId and userId associated with it, that's enough information to do any kind of lookup that we would need to do, so the comment's _id is now unnecessary. Thanks for being patient with me on this, I guess these things happen when you change things a million times during the dev process like I did lol. Will push up a change in a bit that will get rid of this issue entirely and further simplify the Rating collection @carpenterd777

Reply:
Yeah, that actually makes sense too - and like you said that also sidesteps the whole discussion above. I'll take a look once its pushed up

Reply:
>  it just needs to be a string
yes